battle/src/battle.rs

```diff
@@ -52,18 +58,40 @@ impl Battle {
         }
 
         self.replay.updates.push_back((p1, p2));
-
-        let p1_events = self.player_1.update(p1, &mut self.p1_rng, &mut self.garbage_rng);
-        let p2_events = self.player_2.update(p2, &mut self.p2_rng, &mut self.garbage_rng);
+        
+        let p1_events = self.player_1.update(
+            p1,
+            &mut self.p1_rng,
+            &mut self.garbage_rng,
+            self.turn_based && !self.p1_turn

```

Comment:
This condition reads weirdly. player 1 update gets passed true if... it's not player 1's turn? You might want to flip the meaning of the parameter.